### ISSUE TITLE
OASFactory.INSTANCE ignores the resolver set

### ISSUE THREAD

**Issue by johnament** [eclipse/microprofile-open-api#80]
If you set the resolver via `OASFactoryResolver.setInstance`, that resolver is not used by invocations to the factory.  The value is set when the class is loaded.

Instead, you should just ask for the resolved instance on each invocation.  I also wonder if this class should be an API, with methods defined, instead of being a static factory?  Maybe a different issue.

**Comment by arthurdm** [comment]
@leochr / @mrglavas - can you please take a look into this one?

**Comment by leochr** [comment]
The OASFactory follows the same static resolver pattern as of [ConfigProvider](https://github.com/eclipse/microprofile-config/blob/master/api/src/main/java/org/eclipse/microprofile/config/ConfigProvider.java) from MP Config. 

We also posted in the google forum about the static resolver pattern and were told that it's fine/reasonable to expect that there is only one implementation of resolver within a vendor's environment and it be set at the beginning (vendor's responsibility). `OASFactoryResolver.setInstance` is to be used by OSGi environment while service loader pattern is not supported. Vendor should set the factory resolver before the factory class is loaded. 

@Emily-Jiang your thoughts about the static resolver?

We decided the OASFactory to be static so it's easy for end users to invoke the static createObject method. e.g. `OASFactory.createObject(Info.class).title("Airlines")`
